### ISSUE TITLE
SimpleInvalidationHandler  have a leaks of memory

### ISSUE THREAD

**Issue by Larnoo** [osmdroid/osmdroid#390]
MapView:
****
Constructor

mTileRequestCompleteHandler = tileRequestCompleteHandler == null
				? new SimpleInvalidationHandler(this)
				: tileRequestCompleteHandler;
		mTileProvider = tileProvider;
		mTileProvider.setTileRequestCompleteHandler(mTileRequestCompleteHandler);
****
@Override
	protected void onDetachedFromWindow() {
		this.mZoomController.setVisible(false);
		this.onDetach();
		super.onDetachedFromWindow();
	}
****
public void onDetach() {
		this.getOverlayManager().onDetach(this);
		mTileProvider.detach();
	}
****

SimpleInvalidationHandler ：

****
public class SimpleInvalidationHandler extends Handler {
	private final View mView;

	public SimpleInvalidationHandler(final View pView) {
		super();
		mView = pView;
	}

	@Override
	public void handleMessage(final Message msg) {
		switch (msg.what) {
		case MapTile.MAPTILE_SUCCESS_ID:
			mView.invalidate();
			break;
		}
	}
}
****
The default Handler of mTileRequestCompleteHandler, You are not set mView as null when onDetachedFromWindow

**Comment by spyhunter99** [comment]
Great catch.
